### PR TITLE
[WIP]: Improve events/v1 validation

### PR DESCRIPTION
**Pull request by dgrisonnet**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:

Correct a few incorrectness in the current events validator to match the behavior defined in the KEP.

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/apis/core/validation/events.go [hunk 1]
```diff
@@ -68,6 +67,9 @@ func ValidateEventCreate(event *core.Event, requestVersion schema.GroupVersion)
 	if event.Source.Component != "" || event.Source.Host != "" {
 		allErrs = append(allErrs, field.Invalid(field.NewPath("source"), "", "needs to be unset"))
 	}
+	if event.Series != nil {
+		allErrs = append(allErrs, field.Invalid(field.NewPath("series"), event.Series, "needs to be nil when an event is created"))
```
Comment (dgrisonnet) [inline pkg/apis/core/validation/events.go:71]: Theoretically, we could allow the creation of Events with a Series but that wouldn't match the behavior defined in the KEP which doesn't sound correct. IMHO this validation should gate for correct utilization of the API by the clients.

File: pkg/apis/core/validation/events.go [hunk 2]
```diff
@@ -93,7 +95,6 @@ func ValidateEventUpdate(newEvent, oldEvent *core.Event, requestVersion schema.G
 	allErrs = append(allErrs, ValidateImmutableField(newEvent.FirstTimestamp, oldEvent.FirstTimestamp, field.NewPath("firstTimestamp"))...)
 	allErrs = append(allErrs, ValidateImmutableField(newEvent.LastTimestamp, oldEvent.LastTimestamp, field.NewPath("lastTimestamp"))...)
 	allErrs = append(allErrs, ValidateImmutableField(newEvent.Count, oldEvent.Count, field.NewPath("count"))...)
-	allErrs = append(allErrs, ValidateImmutableField(newEvent.Reason, oldEvent.Reason, field.NewPath("reason"))...)
```
Comment (dgrisonnet) [inline pkg/apis/core/validation/events.go:96]: this was duplicated

**Comment by dgrisonnet** [comment]
I will look into that again once my other Event's related effort are merged.

/remove-lifecycle stale

**Comment by dims** [comment]
This PR has the label work-in-progress, please revisit to see if you still need this, please close if not

**Pull request closed without merging**